File: perl/pe025.pl
#!/usr/bin/perl -w
use strict;

use Math::BigInt;
use Memoize;
memoize('fib');

my $i = 0;
my $x = Math::BigInt->new;
while ($x->length() < 1000) {
  $i++;
  $x = Math::BigInt->new(fib($i));
}
print $i,"\n";

sub fib {
  my $i = shift;
  return $i if ($i < 2);
  return Math::BigInt->new(fib($i-2) + fib($i-1));
}

File: ruby/pe025.rb
#!/usr/bin/ruby

def fib(n, vector)
  if n < 2
    vector.push(n)
  else
    vector.push(vector[n - 2] + vector[n - 1])
  end
end

limit = 1000
i = 0
vector = Array.new
fib(i, vector)
while vector[i].to_s.length < limit
  i += 1
  fib(i, vector)
end

puts(i)
